eng/scripts/update-pathversions.rs

```diff
@@ -10,14 +10,24 @@ toml_edit = "^0.22"
 ---
 
 
+
 use regex::Regex;
 use std::io::Write;
 use std::{env, error::Error, fs, path::PathBuf};
 use toml_edit::{value, DocumentMut, Item, Table};
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let script_root = PathBuf::from(env::var("CARGO_MANIFEST_DIR")?);
-    let repo_root = script_root.join("../../..").canonicalize()?;
+    let mode = env::args().nth(1).unwrap_or("update".to_string());
+
+    // in update mode, ignore any depdendencies that don't already have a version
+    // add mode will not add versions to packages that are `publish = false` or to dev-dependencies
+    if mode != "add" && mode != "update" {
+        panic!("Invalid mode. Use 'add' or 'update'.")
+    }
+
+    //let script_root = PathBuf::from(env::var("CARGO_MANIFEST_DIR")?);
+    //let repo_root = script_root.join("../../..").canonicalize()?;
+    let repo_root = PathBuf::from("/home/pahallis/repos/azure-sdk-for-rust").canonicalize()?;
```

Comment:
Me nor the hosted agents are going to have this directory. Recommend you look up the tree until you find a file system item (don't check if it's a file or dir) for ".git". See my `update_cratenames.rs` script - the function at the bottom. Can look for `Cargo.toml` as well. Could see if the `file!()` macro works in a cargo script to get the script file itself, but it might be relocated. I've never checked.

Reply:
😬I really thought I had pulled my debug stuff out.  fixed

Reply:
CARGO_MANIFEST_DIR is the folder containing the script